rfcs/20220310-hip-support/README.md

```diff
@@ -0,0 +1,172 @@
+# AMD backend support (RFC)
+
+## Introduction
+The idea for this RFC is to expose the AMD backend for oneDNN.
+- This branch introduces HIP backend support for the primitives that are currently supported for CUDA backend.
```

Comment:
My understanding of the ultimate goal is that all oneDNN functionality will be implemented for AMD GPUs in two steps:
1. A subset of oneDNN functionality will be covered via MIOpen library. The subset doesn't necessarily have to match the subset covered via cuBLAS and cuDNN for NVIDIA
2. The remaining functionality will be implemented via SYCL 

The goal of this RFC seems to be covering the first part. Because of that I'd expect this RFC to cover the following:
1. Design - skeleton code has to be provided and described so that it was clear what architecture of the primitives will look like
2. Coverage - we need to clearly understand what is going to be covered via MIOpen and what is going to be covered via SYCL. I suggest to create a table for each primitive that would contain all cases that the primitive supports for Intel hardware so that we can track coverage. For example, the table should contain combinations of data types, post-ops, scales, propagation kinds, formats (NCHW, NHWC, etc), etc. You can get it from oneDNN documentation
4. Targeting memory model (SYCL buffers, USM)
5. Validation (e.g. HW used for validation)
6. System requirements - versions DPC++ compiler, HIP, MIOpen, etc